feat(wip): change Logos-open-url to send messages over dbus

ctrlaltf24 opened the pull request:
Created a tiny rust binary to send messages over dbus, then implemented a dbus server in python

Also refactored startup to initialize the python object then call start to enable being able to use the app instance in the dbus code.

TODO:
- [x] Still need to package and extract the binary
- [ ] double check dbus naming conventions to make sure the naming makes sense- much easier to change before we release anything
- [ ] test snap
- [ ] test flatpak
- [ ] test gui/tui exit conditions, need to make sure bases are covered after refactor
- [ ] test what happens when launching OD multiple times
- [ ] test opening link in the browser before OD starts - can we configure our .desktop files to open automatically?


Tested:
- install and run using chromium - works.

ctrlaltf24 committed:
fix: ensure wine appimage is downloaded

Not sure why this didn't pop up earlier in testing, but for some reason it appeared on my machine so added an additional check to make sure the appimage downloads

ctrlaltf24 committed:
feat(wip): change Logos-open-url to spawn a tiny process

Created a tiny rust binary to send messages over dbus, then implemented a dbus server in python

Still need to package and extract the binary and double check dbus naming conventions to make sure the naming makes sense- much easier to change before we release anything

Also refactored startup to initialize the python object then call start to enable being able to use the app instance in the dbus code.

Testing still needed as of time of writing:
- snap/flatpak
- gui/tui exit conditions, need to make sure bases are covered after refactor
- launching OD multiple times
- opening link in the browser before OD starts - can we configure our .desktop files to open automatically?
- consider a --daemon cli option in case someone just wants to run the dbus server in the background so they can open Logos urls all the time. Perhaps we can do this later - not very important, they can always have Logos open first before they hit the link (we can wait until it's a requested feature)

Tested:
- install and run using chromium - works.

thw26 commented on ou_dedetai/dbus.py:
```diff
@@ -0,0 +1,25 @@
+        self.app = app
+
+    @method()
+    def Launch(self, args: 'as') -> None: # type: ignore[valid-type]
```
nit: comment should have two preceding spaces

thw26 reviewed (commented):
Did an initial code review. Still needs tested.

ctrlaltf24 committed:
fix: completed install process

Tested:
- built binary
- run uninstall, then install from binary
- ran logos from binary
- logged in using the browser
- observed Logos application logged in

ctrlaltf24 committed:
Merge remote-tracking branch 'upstream/main' into feat-implement-dbus-forwarder
